### PR TITLE
Perik k 50 create delete user endpoint

### PR DESCRIPTION
**Pull request by PerikK**

### CODE REVIEW

**Review by homonoviscoding — DISMISSED** [review]

File: src/controllers/user.js [hunk 1]
```diff
@@ -83,3 +85,39 @@ export const updateById = async (req, res) => {
 
   return sendDataResponse(res, 200, { user: updatedUser })
 }
+
+export const deleteUserById = async (req, res) => {
+  const id = Number(req.params.id)
```
Comment (angustownsley) [inline src/controllers/user.js:90]: What happens if this is not a number? We should probably create a response if that is the case

File: src/routes/user.js [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
+  validateAuthentication,
+  validateTeacherRole
+} from '../middleware/auth.js'
 
 const router = Router()
 
 router.post('/', create)
 router.get('/', validateAuthentication, getAll)
 router.get('/:id', validateAuthentication, getById)
+router.patch('/:id', validateAuthentication, validateTeacherRole, updateById)
```
Comment (angustownsley) [inline src/routes/user.js]: you can delete this line, the other patch route is the correct one

**Review by angustownsley — CHANGES_REQUESTED** [review]

File: src/controllers/user.js [hunk 3]
```diff
@@ -90,11 +90,12 @@ export const deleteUserById = async (req, res) => {
   const id = Number(req.params.id)
   const canDelete = validateCanModify(req)
 
+  if (isNaN(id)) {
+    return sendDataResponse(res, 400, ERR.BAD_REQUEST)
```
Comment (tzoltie) [inline src/controllers/user.js]: this should be `sendDataResponse(res, 400, { error: ERR.BAD_REQUEST } `

File: src/controllers/deliveryLog.js [hunk 4]
```diff
@@ -8,7 +8,10 @@ export const createDeliveryLog = async (req, res) => {
   const { firstName, lastName } = req.user
 
   if (!cohortId || !lines) {
-    return sendDataResponse(res, 400, { error: ERR.INCOMPLETE_REQUEST })
+    return res.status(400).json({
+      status: 'error',
+      data: ERR.INCOMPLETE_DELIVERY_LOG_REQUEST
+    })
```
Comment (angustownsley) [inline src/controllers/deliveryLog.js]: Change to use sendDataResponse

File: src/controllers/user.js [hunk 5]
```diff
@@ -87,29 +89,31 @@ export const updateById = async (req, res) => {
 
 export const deleteUserById = async (req, res) => {
   const id = Number(req.params.id)
-  const user = req.user
-  if (!id) {
-    return res.status(400).json({
+  const canDelete = validateCanModify(req)
+
+  if (!canDelete) {
+    return res.status(401).json({
```
Comment (tzoltie) [inline src/controllers/user.js]: this error response body should use the `sendDataResponse` function

**Review by angustownsley — CHANGES_REQUESTED** [review]

**Review by tzoltie — DISMISSED** [review]

**Review by angustownsley — DISMISSED** [review]

File: src/controllers/user.js [hunk 6]
```diff
@@ -83,3 +85,37 @@ export const updateById = async (req, res) => {
...
+
+    await dbClient.post.deleteMany({
+      where: { userId: id }
+    })
+    await dbClient.profile.delete({
+      where: { userId: id }
+    })
+
+    await User.deleteUserByIdDb(userToDelete.id)
+    return sendDataResponse(res, 201, { deleted_user: userToDelete })
```
Comment (Auenc) [inline src/controllers/user.js]: 200 is the correct response for this

**Review by Auenc — CHANGES_REQUESTED** [review]

**Review by angustownsley — APPROVED** [review]

**Review by Auenc — APPROVED** [review]

**Pull request merged**